pkg/config/remote/client/client.go

```diff
@@ -39,12 +39,14 @@ const (
 	maxMessageSize = 1024 * 1024 * 110 // 110MB, current backend limit
 )
 
-// ConfigUpdater defines the interface that an agent client uses to get config updates
-// from the core remote-config service
-type ConfigUpdater interface {
+// ConfigFetcher defines the interface that an agent client uses to get config updates
+type ConfigFetcher interface {
 	ClientGetConfigs(context.Context, *pbgo.ClientGetConfigsRequest) (*pbgo.ClientGetConfigsResponse, error)
 }
 
+// fetchConfigs defines the function that an agent client uses to get config updates
+type fetchConfigs func(context.Context, *pbgo.ClientGetConfigsRequest, ...grpc.CallOption) (*pbgo.ClientGetConfigsResponse, error)
+
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**💭 Thought:** ‏It's a bit annoying to have to write the method signature twice here, but I'm not sure we can use `fetchConfigs` in the interface.

Reply:
While they are similar signatures, they represent two different thing that I don't think we necessarily want to unify